R/aaa_survival_prob.R

```diff
@@ -145,3 +145,142 @@ matrix_to_nested_tibbles_survival <- function(x, time) {
   dplyr::group_nest(res, .row, .key = ".pred")$.pred
 }
 
+
+
+# summary_survfit helpers -------------------------------------------------
+
+
+survfit_summary_typestable <- function(object){
+  # make matrix of dimension n_times x n_obs
+  sanitize_element <- function(x) {
+    if (!is.matrix(x)) {
+      x <- matrix(x, ncol = length(object$n))
+    }
+    x
+  }
+  # sanitize elements we care about
+  elements <- available_survfit_summary_elements(object)
+  for (i in elements) {
+    object[[i]] <- sanitize_element(object[[i]])
+  }
+
+  object
+}
+
+available_survfit_summary_elements <- function(object) {
+  intersect(
+    names(object),
+    c("surv", "std.err", "lower", "upper", "cumhaz", "std.chaz")
```

Comment:
we currently don't have a prediction type for cumulative hazard but this should make it a lot easier to enable that in future